README.md

```diff
@@ -1,2 +1,114 @@
+
+}
+// calls to GlobalSpanManager.activeSpan() here will receive the active span from before the try block.
+````
+Due to the `try-with-resources` construct, the span will automatically close and restore the _active span_ state to what it was previously.  
+It should be noted that `Tracer` implementations need not be concerned with this functionality;
+it is provided by the `GlobalSpanManager` being invoked from the global tracer.
+
+### From inbound filters at the application boundary (server)
+To create a generic _opentracing_ compatible filter, the hypothetical filter could look similar to
+the following fragment:
+````java
+public void doFilter(someContext, someFilterChain) {
+    // Obtain the inbound context:
+    Format<Carrier> someFormat = ... // usually known by filter
+    Carrier someCarrier = ... // Usually obtained from someContext
+    SpanContext inboundContext = GlobalTracer.tracer().extract(someFormat, someCarrier);
+    
+    // Create a new span as child of the inbound context.
+    try (Span span = GlobalTracer.tracer().buildSpan("inboundOperation")
+            .asChildOf(inboundContext).start()) {
+        someFilterChain.filter(someContext); // continue the filtered operation within 'span'
+    } finally { // Clears any active spans.
+        // This is mostly a safety measure that can be taken in outermost filters
+        // due to the use of threadpools on most application platforms.
+        // clearActiveSpans() can be called as cleanup before returning a thread back to the pool.
+        GlobalSpanManager.clearActiveSpans();
+    }
+}
+````
+
+### From outbound filters at the application boundary (client)
+The global tracer can also be used to inject the _active span_'s context into outbound requests:
+````java
+// Depending on the type of client, gather the format and carrier to propagate SpanContext with.
+Format<Carrier> someFormat = ...
+Carrier someCarrier = ...
+// Leave the injection to the actual tracer implementation:
+GlobalTracer.tracer().inject(GlobalSpanManager.activeSpan(), someFormat, someCarrier);
+// Call the actual client with the injected SpanContext in the carrier.
+````
+
+## Default in-process propagation
+By default, starting a new java `Thread` will lose the _active span_ from this library.
+Therefore there are several utility classes provided in the `global.concurrent` subpackage:
+ 1. `TracedRunnable.of(delegate)` which will capture the _active span_ when it is invoked to be re-activated 
+    just before calling the delegate `Runnable` implementation,
+    making sure to deactivate it again after the call has finished.
+ 2. `TracedCallable.of(delegate)` which will capture the _active span_ when it is invoked to be re-activated
+    just before calling the delegate `Callable` implementation,
+    making sure to deactivate it again after the call has finished.
+ 3. `TracedExecutorService.traced(delegate)` which will wrap a given `ExecutorService`
```

Comment:
I think the desc can be tightened to ..

traced runnable or callables will run with the active span of the submitting thread.

or.. something